lib/npg_qc/autoqc/checks/rna_seqc.pm

```diff
@@ -0,0 +1,328 @@
+        my $rpt_dir = $rp_dir . $self->tag_label;
+        $out_dir = join q[/], $archive_qc_rna_seqc_path, $rp_dir, $rpt_dir;
+    }
+    return $out_dir;
+}
+
+has 'input_str' => (is => 'ro',
+                    isa        => 'Str',
+                    lazy_build => 1,);
+
+sub _build_input_str {
+    my ($self) = @_;
+    my $sample_id = $self->lims->sample_id;
+    my $library_name = $self->lims->library_name // $sample_id;
+    my @library_names = split q[ ], $library_name;
+    my $input_file = $self->bam_file;
+    return qq["$library_names[0]|$input_file|$sample_id"];
+}
+
+has 'reference_fasta' => (is => 'ro',
+                          isa => 'Maybe[Str]',
+                          lazy_build => 1,);
+
+sub _build_reference_fasta {
+    my ($self) = @_;
+    return $self->refs->[0];
+}
+
+has 'bam_file' => (is         => 'ro',
+                   isa        => 'Str',
+                   lazy_build => 1,);
+
+sub _build_bam_file {
+    my $self = shift;
+    return $self->input_files->[0];
+}
+
+has 'transcriptome' => (is         => 'ro',
+                        isa        => 'Maybe[Str]',
+                        lazy_build => 1,);
+
+sub _build_transcriptome {
+    my $self = shift;
+    my $trans_gtf = $self->rnaseqc_gtf_file;
+    return $trans_gtf;
+}
+
+sub _command {
+    my ($self) = @_;
+    my $out_dir = $self->rna_seqc_path;
+    # check existence of RNA_SeQC's output directory create if it doesn't
+    if ( ! -d $out_dir) {
+        make_path($out_dir);
+    }
+    my $single_end_option=q[];
+    if(!npg::api::run->new({id_run => $self->id_run})->is_paired_read()){
+        $single_end_option=q[-singleEnd];
+    }
+    my $command = $self->java_cmd. sprintf q[ -Xmx%s -XX:%s -XX:%s -jar %s -s %s -o %s -r %s -t %s -ttype %d %s],
+                                           $self->java_max_heap_size,
+                                           $self->java_gc_type,
+                                           $self->java_use_perf_data,
+                                           $self->java_jar_path,
+                                           $self->input_str,
+                                           $out_dir,
+                                           $self->reference_fasta,
+                                           $self->transcriptome,
+                                           $self->transcript_type,
+                                           $single_end_option;
+    return $command;
+}
+
+has 'config_file_loc' => (is         => 'ro',
+                          isa        => 'Str',
+                          lazy_build => 1,);
+
+override 'can_run' => sub {
+    my $self = shift;
+    my $l = $self->lims;
+    my $can_run = 1;
+    my @comments;
+    if(! $self->alignments_in_bam) {
+        push @comments, q[Alignments_in_bam is false];
+        $can_run = 0;
+    }
+    if ((! $l->library_type || $l->library_type !~ /(?:cD|R)NA/sxm) && $l->library_type ne q[Pre-quality controlled]) {
```

Comment:
The intention of multiple if's instead of using elsif is to store in @comments *all* the reasons for the check not to be run, so all of them are fixed at the same time - if that's the case, as opposed to saving only the first one, (fix it?), and potentially have another skip in subsequent attempts.